iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/file_lock.hpp

```diff
@@ -48,27 +50,30 @@ enum class FileLockError
 ///        if the process crashes with a segfault or using SIGKILL. 'lslocks' can be used to display all system-wide
 ///        locks (see man page)
 /// @code
-///    iox::posix::FileLock::create(nameOfmyLock)
-///        .and_then([] { std::cout << "We aquired the lock!" << std::endl; })
-///        .or_else([](auto& error) {
-///            if (error == FileLockError::LOCKED_BY_OTHER_PROCESS)
-///            {
-///                std::cout << "Some other process is running and holds the lock!" << std::endl;
-///            }
-///        });
+///   auto fileLock = iox::posix::FileLockBuilder().name("myLockName")
+///                                                .path("/tmp")
+///                                                .permission(iox::cxx::perms::owner_all)
+///                                                .create()
+///                                                .expect("Oh no I couldn't create the lock");
 /// @endcode
-class FileLock : public DesignPattern::Creation<FileLock, FileLockError>
+class FileLock
 {
   public:
-    static constexpr int32_t ERROR_CODE = -1;
     static constexpr int32_t INVALID_FD = -1;
     static constexpr const char LOCK_FILE_SUFFIX[] = ".lock";
     static constexpr uint64_t FILENAME_LENGTH = platform::IOX_MAX_FILENAME_LENGTH
                                                 - sizeof(platform::IOX_LOCK_FILE_PATH_PREFIX) / sizeof(char)
                                                 - sizeof(LOCK_FILE_SUFFIX) / sizeof(char);
 
+    /// @brief A file name without any containing slash (path separator)
+    ///        For instance "myLock.lock"
     using FileName_t = cxx::string<FILENAME_LENGTH>;
-    using PathName_t = cxx::string<platform::IOX_MAX_PATH_LENGTH>;
+    /// @brief The full path to the file, including the file
+    ///        For instance "/path/to/myLock.lock"
+    using FilePath_t = cxx::string<platform::IOX_MAX_PATH_LENGTH>;
+    /// @brief The directory to the file
+    ///        For instance "/path/to/"
+    using PathName_t = cxx::string<platform::IOX_MAX_PATH_LENGTH - FILENAME_LENGTH - 1>;
```

Comment:
I already wrote a lengthy comment that there is a off by one error if `m_path` uses the full size and has not path separator and `m_name` also uses the full size. Then I noticed that the `-1` is not for null termination but for the path separator.

Please use a `constexpr` ... and you owe me 2 minutes of my life :laughing:

Reply:
I will ask Hypnotoad if it grants you two additional minutes in your life.